pylab/accounts/tests/test_settings.py

```diff
@@ -47,3 +49,17 @@ def test_user_profile_locale_middleware(self):
         resp = self.app.get('/accounts/settings/', user='u1')
         # Website interface is displayed in lithuanian
         self.assertTrue(b'Saugoti' in resp.content)
+
+    def test_requirement_to_accept_terms_of_service(self):
+        auth_models.User.objects.create_user('u2_not_accepted_terms_yet')
+        resp = self.app.get('/accounts/settings/', user='u2_not_accepted_terms_yet')
+        resp.form['language'] = 'en'
+        resp = resp.form.submit()
+        # Don't allow to save unless user accepts terms of service
+        self.assertEqual(resp.status_int, 200)
+        self.assertTrue(b'has-error' in resp.content)
```

Comment:
A css or xpath query to a particular field would be much better. Now any field can have `has-error` and this might give false positive.